transform_dialect/examples/accel/mmt4d_codegen_spec.mlir

```diff
@@ -0,0 +1,117 @@
+// CHECK-IR:   memref.subview {{.*}} : memref<32x384x8x1xf32, strided<[3072, 8, 1, 1], offset: ?>> to memref<32x8x8x1xf32, strided<[3072, 8, 1, 1], offset: ?>>
+// CHECK-IR:   linalg.generic {indexing_maps = [#map2, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins{{.*}} outs(%alloca_1 : memref<16x8x8x1xf32>) { linalg.yield }
+// CHECK-IR:   linalg.generic {indexing_maps = [#map2, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins{{.*}} outs(%alloca_0 : memref<32x8x8x1xf32) { linalg.yield }
+// CHECK-IR:   gpu.thread_id  x
+// CHECK-IR:   gpu.thread_id  y
+// CHECK-IR:   gpu.thread_id  z
+// CHECK-IR:   affine.apply #map3(%6)
+// CHECK-IR:   affine.apply #map3(%5)
+// CHECK-IR:   memref.subview %alloca_1 : memref<16x8x8x1xf32> to memref<8x8x8x1xf32, strided<[64, 8, 1, 1], offset: ?>>
+// CHECK-IR:   memref.subview %alloca_0 : memref<32x8x8x1xf32> to memref<8x8x8x1xf32, strided<[64, 8, 1, 1], offset: ?>>
+// CHECK-IR:   memref.subview {{.*}} : memref<16x32x8x8xf32, strided<[4096, 64, 8, 1], offset: ?>> to memref<8x8x8x8xf32, strided<[4096, 64, 8, 1], offset: ?>>
+// CHECK-IR:   linalg.generic {indexing_maps = [#map2, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins{{.*}} outs(%alloca : memref<8x8x8x8xf32>) { linalg.yield }
+// CHECK-IR:   linalg.mmt4d ins({{.*}}, {{.*}} : memref<8x8x8x1xf32, strided<[64, 8, 1, 1], offset: ?>>, memref<8x8x8x1xf32, strided<[64, 8, 1, 1], offset: ?>>) outs(%alloca : memref<8x8x8x8xf32>)
+// CHECK-IR:   linalg.generic {indexing_maps = [#map2, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%alloca : memref<8x8x8x8xf32>) outs({{.*}} : memref<8x8x8x8xf32, strided<[4096, 64, 8, 1], offset: ?>>) { linalg.yield }
+// CHECK-IR:   gpu.barrier
+// CHECK-IR:  }
+// CHECK-IR:  return
+// CHECK-IR: }
+
+module attributes { transform.with_named_sequence } {
+  transform.named_sequence @cleanup(%variant_op: !transform.any_op {transform.readonly}) {
+    %func = transform.structured.match ops{["func.func"]} in %variant_op : (!transform.any_op) -> !transform.any_op
+      transform.apply_patterns to %func {
+      transform.apply_patterns.linalg.tiling_canonicalization
+      transform.apply_patterns.iree.fold_fill_into_pad
+      transform.apply_patterns.scf.for_loop_canonicalization
+      transform.apply_patterns.canonicalization
+    } : !transform.any_op
+    transform.iree.apply_licm %func : !transform.any_op
+    transform.iree.apply_cse %func : !transform.any_op
+    transform.yield
+  }
+
+  transform.sequence failures(propagate) {
+  ^bb1(%variant_op: !transform.any_op):
+    %matmul = transform.structured.match ops{["linalg.mmt4d"]} in %variant_op : (!transform.any_op) -> !transform.any_op
+
+    // First level tile to forall with tile_sizes [16, 32].
+    %forall, %tiled_matmul =
+      transform.structured.tile_to_forall_op %matmul tile_sizes [16, 32]
```

Comment:
Can you make this `[2, 4]`.. That would effectively do a movement of data needed for `2x4` tiles of inner tiles size (which is `8x8x8` as I said below`.